### PR TITLE
Pin cuTENSOR

### PR DESCRIPTION
**Pull request by leofang**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

cc: @jakirkham

### CODE REVIEW

**Review by kkraus14 — APPROVED** [review]

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -364,6 +364,8 @@ cairo:
   - 1.16
 cfitsio:
   - 3.470
+cutensor:
+  - 1.2.2.5
```
Comment (isuruf) [inline recipe/conda_build_config.yaml:368]: This should match the `run_exports` version
```suggestion
  - 1
```

Comment (leofang) [reply recipe/conda_build_config.yaml:368]: Hmmm...I don't know if cuTENSOR provides any guarantee, and I'd prefer to follow NCCL's pinning. Sounds like I should fix `pin_subpackage` in cuTENSOR's recipe first?

Comment (isuruf) [reply recipe/conda_build_config.yaml:368]: > Sounds like I should fix pin_subpackage in cuTENSOR's recipe first?

Yes, but I would go with the current run_exports as the SONAME only has the major version and I'd rather we trust the upstream developers to bump the SONAME if they make a breaking change.

Comment (leofang) [reply recipe/conda_build_config.yaml]: Make sense, thanks @isuruf! Applied your patch.

**Comment by leofang** [comment]
Thanks, @isuruf @kkraus14!

**Comment by jakirkham** [comment]
Thanks Leo! 😄

**Pull request merged**